python/cuml/experimental/common/base.pyx

```diff
@@ -0,0 +1,535 @@
+                # check output format
+                self._check_output_type(data)
+                # train logic goes here
+
+            def predict(self, data, ...):
+                # check output format
+                self._check_output_type(data)
+                # inference logic goes here
+
+            def get_param_names(self):
+                # return a list of hyperparam names supported by this algo
+
+        # stream and handle example:
+
+        stream = cuml.cuda.Stream()
+        handle = cuml.Handle(stream=stream)
+
+        algo = MyAlgo(handle=handle)
+        algo.fit(...)
+        result = algo.predict(...)
+
+        # final sync of all gpu-work launched inside this object
+        # this is same as `cuml.cuda.Stream.sync()` call, but safer in case
+        # the default stream inside the `cumlHandle` is being used
+        base.handle.sync()
+        del base  # optional!
+    """
+
+    def __init__(self, *,
+                 handle=None,
+                 verbose=False,
+                 output_type=None):
+        """
+        Constructor. All children must call init method of this base class.
+
+        """
+        self.handle = raft.common.handle.Handle() if handle is None \
+            else handle
+
+        # Internally, self.verbose follows the spdlog/c++ standard of
+        # 0 is most logging, and logging decreases from there.
+        # So if the user passes an int value for logging, we convert it.
+        if verbose is True:
+            self.verbose = logger.level_debug
+        elif verbose is False:
+            self.verbose = logger.level_info
+        else:
+            self.verbose = verbose
+
+        self.output_type = _check_output_type_str(
+            cuml.global_settings.output_type
+            if output_type is None else output_type)
+        self._input_type = None
+        self.target_dtype = None
+        self.n_features_in_ = None
+
+        nvtx_benchmark = os.getenv('NVTX_BENCHMARK')
+        if nvtx_benchmark and nvtx_benchmark.lower() == 'true':
+            self.set_nvtx_annotations()
+
+    def __repr__(self):
+        """
+        Pretty prints the arguments of a class using Scikit-learn standard :)
+        """
+        cdef list signature = inspect.getfullargspec(self.__init__).args
+        if len(signature) > 0 and signature[0] == 'self':
+            del signature[0]
+        cdef dict state = self.__dict__
+        cdef str string = self.__class__.__name__ + '('
+        cdef str key
+        for key in signature:
+            if key not in state:
+                continue
+            if type(state[key]) is str:
+                string += "{}='{}', ".format(key, state[key])
+            else:
+                if hasattr(state[key], "__str__"):
+                    string += "{}={}, ".format(key, state[key])
+        string = string.rstrip(', ')
+        return string + ')'
+
+    def dispatch_func(self, func_name, original_func, *args, **kwargs):
```

Comment:
Add docstring to the function describing how it works, and for what estimators it can work and when it would need to be overriden

Reply:
Docstring needs documentation of the parameters of the function